cfme/cloud/provider/openstack.py

```diff
@@ -24,28 +27,21 @@ class OpenStackProvider(CloudProvider):
     _canvas_element = '//*[@id="noVNC_canvas"]'
     _ctrl_alt_del_xpath = '//*[@id="sendCtrlAltDelButton"]'
 
-    def __init__(self, name=None, endpoints=None, zone=None, key=None, hostname=None,
-                 ip_address=None, api_port=None, api_version=None, sec_protocol=None,
-                 amqp_sec_protocol=None, keystone_v3_domain_id=None, tenant_mapping=None,
-                 infra_provider=None, appliance=None):
-        super(OpenStackProvider, self).__init__(name=name, endpoints=endpoints,
-                                                zone=zone, key=key, appliance=appliance)
-        self.hostname = hostname
-        self.ip_address = ip_address
-        self.api_port = api_port
-        self.api_version = api_version
-        self.keystone_v3_domain_id = keystone_v3_domain_id
-        self.infra_provider = infra_provider
-        self.sec_protocol = sec_protocol
-        self.tenant_mapping = tenant_mapping
-        self.amqp_sec_protocol = amqp_sec_protocol
+    api_port = attr.ib(default=None)
+    api_version = attr.ib(default=None)
+    sec_protocol = attr.ib(default=None)
+    amqp_sec_protocol = attr.ib(default=None)
+    keystone_v3_domain_id = attr.ib(default=None)
+    tenant_mapping = attr.ib(default=None)
+    infra_provider = attr.ib(default=None)
 
+    # todo: move it to collections later
     def create(self, *args, **kwargs):
         # Override the standard behaviour to actually create the underlying infra first.
         if self.infra_provider:
             self.infra_provider.create(validate_credentials=True, validate_inventory=True,
                                        check_existing=True)
-        if self.appliance.version >= "5.6" and 'validate_credentials' not in kwargs:
+        if 'validate_credentials' not in kwargs:
```

Comment:
**Optional**

Just squash this down since its for a version that's no longer strictly supported